src/spatialdata/_core/spatialdata.py

```diff
@@ -962,195 +908,564 @@ def transform_to_coordinate_system(
+                "more elements in the SpatialData object. Deleting the data would corrupt the SpatialData object."
+            )
+
+        # delete the element
+        store = parse_url(self.path, mode="r+").store
+        root = zarr.group(store=store)
+        root[element_type].pop(element_name)
         store.close()
 
-        # old code to support overwriting the backing file
-        # if target_path is None:
-        #     self.path = str(file_path)
-        # else:
-        #     self.path = str(tmp_zarr_file)
-        self._path = Path(file_path)
-        try:
-            if len(self.images):
-                root.create_group(name="images")
-                # add_image_in_memory will delete and replace the same key in self.images,
-                # so we need to make a copy of the keys. Same for the other elements
-                keys = self.images.keys()
-
-                for name in keys:
-                    elem_group = self._init_add_element(name=name, element_type="images", overwrite=overwrite)
-                    write_image(
-                        image=self.images[name],
-                        group=elem_group,
-                        name=name,
-                        storage_options=storage_options,
-                    )
+        if self.has_consolidated_metadata():
+            self.write_consolidated_metadata()
 
-                    # TODO(giovp): fix or remove
-                    # reload the image from the Zarr storage so that now the element is lazy loaded,
-                    # and most importantly, from the correct storage
-                    # element_path = Path(self.path) / "images" / name
-                    # _read_multiscale(element_path, raster_type="image")
-
-            if len(self.labels):
-                root.create_group(name="labels")
-                # keys = list(self.labels.keys())
-                keys = self.labels.keys()
-
-                for name in keys:
-                    elem_group = self._init_add_element(name=name, element_type="labels", overwrite=overwrite)
-                    write_labels(
-                        labels=self.labels[name],
-                        group=elem_group,
-                        name=name,
-                        storage_options=storage_options,
-                    )
+    def _check_element_not_on_disk_with_different_type(self, element_type: str, element_name: str) -> None:
+        only_on_disk = self.elements_paths_on_disk()
+        for disk_path in only_on_disk:
+            disk_element_type, disk_element_name = self._element_type_and_name_from_element_path(disk_path)
+            if disk_element_name == element_name and disk_element_type != element_type:
+                raise ValueError(
+                    f"Element {element_name} is found in the Zarr store as a {disk_element_type}, but it is found "
+                    f"in-memory as a {element_type}. The in-memory object should have a different name."
+                )
```

Comment:
I think this is a bit tricky, if a user see this error message, what should they do?

Reply:
I added an actionable explanation: "If you want to maintain both objects, please rename the in-memory object. Alternatively, you can rename the element on disk (manually)."